protobuf/traits/button.proto

```diff
@@ -0,0 +1,110 @@
+syntax = "proto3";
+
+package smartcore.traits;
+
+option csharp_namespace = "Smartcore.Traits";
+option go_package = "github.com/smart-core-os/sc-api/go/traits";
+option java_multiple_files = true;
+option java_outer_classname = "ChannelProto";
+option java_package = "dev.smartcore.traits";
+option php_namespace = "Smartcore\\Traits";
+
+import "google/protobuf/field_mask.proto";
+import "google/protobuf/timestamp.proto";
+
+// The state and events from a momentary push-button input.
+// This trait facilitates reactions to basic user input.
+// Both the basic button state (pressed or not) as well as higher-level input events like "double click" are supported.
+service ButtonApi {
```

Comment:
I feel like this trait should be renamed to something more specific. Momentary or something like that maybe. Naming this trait `Button` overlaps too much with toggle buttons, it also goes against the principal of naming traits after their action rather than their device.

Some options:
- push
- momentary
- press